### PR TITLE
[#79] make_versioned_release make_compiled_release library functions.

### PR DESCRIPTION
**Pull request by kindly**
Add functions so they can be used as a library.

### CODE REVIEW

**Comment by kindly** [comment]
The tests are failing as this only works with the 'dev' branch on ocdsmerge.

This will not work on the pypi version on ocdsmerge due to it not accepting  `merge_rules` as an argument.

@jpmckinney I can make the tests work on the master branch of ocdsmerge and make a pypi realease for that? or shall we wait for the 'dev' branch to be finished?

**Comment by jpmckinney** [comment]
I have a few notes of things I want to finish on the dev branch. I'll see if I can do those tomorrow.

File: ocdskit/cli/commands/compile.py [hunk 1]
```diff
@@ -1,10 +1,24 @@
 from collections import defaultdict, OrderedDict
+from functools import lru_cache
 
 import ocdsmerge
 
 from .base import BaseCommand
 
 
+@lru_cache()
```
Comment (jpmckinney) [inline ocdskit/cli/commands/compile.py]: I don't think this works, because dict's are unhashable.

**Comment by jpmckinney** [comment]
I've merged the `dev` branch, but I want to update the readme before making a release.

**Comment by jpmckinney** [comment]
OCDS Merge 0.5 is released. I'll look at this now.

File: ocdskit/cli/commands/compile.py (hunk 1 above)
Comment (jpmckinney) [reply ocdskit/cli/commands/compile.py]: Assuming we want to allow users to pass a dict as the schema (allowed by OCDS Merge 0.5).

**Comment by jpmckinney** [comment]
This conflicts with some changes I made while preparing OCDS Merge 0.5.

However, looking at the implementation in this PR, it looks like Kingfisher would have to take responsibility for collecting releases by OCID, and then calling `make_compiled_release` or `make_versioned_release`, which are very thin wrappers around OCDS Merge. If Kingfisher is taking responsibility for the only new feature that OCDS Kit provides, then it makes more sense for Kingfisher to just call OCDS Merge directly.

Closing as a new PR would be required anyhow to avoid the new merge conflicts.

**Pull request closed without merging**